protos/core.proto

```diff
@@ -260,24 +267,414 @@ message CompiledGraph {
   reserved 5, 6;
 }
 
+message CompiledGraphV3 {
+  message ColumnDescriptor {
+    // The identifier for the column, using multiple parts for nested records.
+    repeated string path = 1;
+
+    // A text description of the column.
+    string description = 2;
+
+    // A list of BigQuery policy tags that will be applied to the column.
+    repeated string bigquery_policy_tags = 3;
+  }
+
+  message Table {
+    // Target is the unique action identifier.
+    Target target = 1;
+
+    // Canonical Target is the unique action identifier, with no target
+    // overrides applied.
+    // This option cannot be manually specified in action configs.
+    Target canonical_target = 2;
+
+    // Targets (non-canonical) of actions that this action is dependent on.
+    repeated Target dependency_targets = 3;
+
+    // Path to the source file that the contents of the action is loaded from.
+    string source = 4;
+
+    // Path that the config of the action is loaded from.
+    // This option cannot be manually specified in action configs.
+    string config_path = 5;
```

Comment:
I'm a bit confused about these two. I *think* `source` would be `foo.sqlx`, `bar.js`, or `baz.sql`, and `config_path` would only be set to `...actions.yaml`. Is that correct?

I'd like to more tightly define this. I'd propose something like:

```
message Source {
  string declaration_file = 1;
  string source_file = 2;
}
```

And I'd propose that *both* fields are *always* set (sometimes - as with SQLX and JS actions - to the same value).

Reply:
Delaying this decision until a later version, as discussed in chat